### DISCUSSION TITLE
How to use Spring Security with graphql-dgs-webflux-starter

### DISCUSSION THREAD

**Discussion by cnolte** [Netflix/dgs-framework#359] [Q&A]
I'm new to DGS so this might be a stupid question. But how do I add Spring ReactiveMethodSecurity to an Endpoint? I was expecting that something simliar like the following works:
```

@DgsComponent
class UsersDataFetcher(private val userService: UserService) {

    @PreAuthorize("isAuthenticated()")
    @DgsQuery
    fun users(): Mono<MutableList<User>> {
        return userService.getAllUsers().collectList()
    }
}
```
This resolves an error: "Can't resolve value (/users) : type mismatch error, expected type LIST".

**Comment by ghost** [comment]
https://netflix.github.io/dgs/advanced/security/

Also, are you returning a Mono<MutableList<User>> instead of a Flux<User> in purpose?

**Reply by cnolte** [reply]
Flux or Mono is not supported as return type

**Reply by willisjtc** [reply]
If i'm getting data from a database with r2dbc, how do I map the Flux/Mono types with the dgs data fetchers?
